pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -214,9 +218,16 @@ func (r *MachineSetSyncReconciler) syncMachineSets(ctx context.Context, mapiMach
 	case machinev1beta1.MachineAuthorityMachineAPI:
 		return r.reconcileMAPIMachineSetToCAPIMachineSet(ctx, mapiMachineSet, capiMachineSet)
 	case machinev1beta1.MachineAuthorityClusterAPI:
+		if capiMachineSet == nil {
+			condErr := r.updateSynchronizedConditionWithPatch(
+				ctx, mapiMachineSet, corev1.ConditionFalse, reasonFailedToGetCAPIMachineSet, errCAPIMachineSetNotFound.Error(), nil)
+
+			return ctrl.Result{}, condErr
+		}
+
```

Comment:
So in this case, the MAPI machine set exists, and the CAPI machine set does not?

And the authority means that the CAPI machine set, that doesn't currently exist, should be authoritative?

The intention here is that the controller creates a CAPI machine set to, become the authoritative API, based on the MAPI machine set, and then, on the next reconcile, things go back to normal

Reply:
I've updated the main `switch` in the reconcile func, and put a comment in. 

I think we should handle this after we've written MAPI -> CAPI as maybe we call a subset of that logic :)